rpcd/playbooks/roles/elasticsearch/files/reindex-liberty.py

```diff
@@ -0,0 +1,306 @@
+    parser.add_argument("--port", default="9200", help="Elasticsearch \
+                        port to connect to (default: 9200)")
+    parser.add_argument("--suffix", default="liberty", help="Suffix for \
+                        reindexed indices. (default: liberty)")
+    parser.add_argument("--clean", action="store_true", help="Clean previousi \
+                        reindexed indices")
+    parser.add_argument("--list", action="store_true", dest="list_indices",
+                        help="List all elasticsearch indices")
+    parser.add_argument("--reindex", action="store_true", help="Reindex all \
+                        legacy indexes")
+    parser.add_argument("--index", default=None, help="Used with --reindex \
+                        to reindex a specific index.")
+    parser.add_argument("--batch", action="store_true", help="Batch \
+                        processing of old indices.  Defaults to oldest index \
+                        through previous days index.")
+    parser.add_argument("--start", help="Start index for batch processing \
+                        (YYYY.MM.DD) (default: beginning of indices)")
+    parser.add_argument("--end", help="End index for batch processing \
+                        (YYYY.MM.DD) (default: the Nth-1 index)")
+    parser.add_argument("--monitor", action="store_true", help="Progress of \
+                        reindexing")
+    parser.add_argument("--continuous", action="store_true", help="Continuous \
+                        monitoring")
+    parser.add_argument("--delay", default=10, type=int, help="Refresh delay \
+                        for continuous monitoring. (default: 10)")
+    parser.add_argument("--verbose", action="store_true", help="Verbose \
+                        output from monitoring functions")
+    parser.add_argument("--drop", action="store_true", help="Drop legacy \
+                        indices")
+    parser.add_argument("--dry-run", action="store_true", dest="dry_run",
+                        help="Prints what will happen for drop, clean and \
+                        batch operations")
+    args = parser.parse_args()
+
+    # Make sure we have hyphens in the correct places
+    if '-' not in list(args.suffix)[0]:
+        args.suffix = '-' + args.suffix
+
+    if args.dry_run:
+        print("Dry run, no operations will be performed.")
+
+    es_host = args.host + ":" + args.port
+    if args.verbose:
+        print("Connecting to elasticsearch at {}").format(es_host)
+    es = Elasticsearch(es_host)
+
+    if args.clean:
+        clean_legacy(es, args)
+    if args.batch:
+        batch(es, es_host, args)
+    if args.list_indices:
+        list_indices(es)
+    if args.reindex:
+        reindex(es, es_host, args)
+    if args.monitor:
+        monitor_reindex(es, args)
+    if args.drop:
+        drop_legacy(es, args)
+    if args.continuous:
+        # TODO(d34dh0r53) I hate this, need to figure out a better way.
```

Comment:
Yeah, I agree and if an RC3 is needed I'll refactor this, in the interest of time it's working and somewhat readable so I think I'll leave it as is for now.